ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/util/DnUtils.java

```diff
@@ -0,0 +1,44 @@
+package org.ccci.idm.user.ldaptive.dao.util;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import org.ccci.idm.user.Dn;
+import org.ldaptive.DnParser;
+import org.ldaptive.LdapAttribute;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class DnUtils {
+    private static final String DELIMITER = ",";
+    private static final String VALUE_DELIMITER = "=";
+
+    @Nonnull
+    public static Dn toDn(@Nullable final String rawDn) {
+        if (rawDn == null) {
+            return Dn.ROOT;
+        }
+
+        final ImmutableList.Builder<Dn.Component> builder = ImmutableList.builder();
+        for (final LdapAttribute attribute : Lists.reverse(DnParser.convertDnToAttributes(rawDn))) {
+            builder.add(new Dn.Component(attribute.getName(), attribute.getStringValue()));
+        }
+        return new Dn(builder.build());
+    }
+
+    @Nonnull
+    public static String toString(@Nonnull final Dn dn) {
+        final StringBuilder sb = new StringBuilder();
+
+        // append components
+        for (final Dn.Component component : Lists.reverse(dn.getComponents())) {
+            if (sb.length() > 0) {
```

Comment:
Curious regarding your preference (which comes up quite frequently in data transformation):

1.Which method name: a. `from<object>(object)` b. `to<object>(other object)` c. `from<object>To<other object>(object)`? 
2. When to add the dto method to the object in question as opposed to an `<object>Utils` class?

The `<` `>` syntax above is meant as a variable, not a Java generic.